client/packages/react-common/src/InstantReactRoom.ts

```diff
@@ -70,12 +71,17 @@ export function useTopicEffect<
     peer: RoomSchema[RoomType]['presence'],
   ) => any,
 ): void {
+  const onEventRef = useRef(onEvent);
+  useEffect(() => {
```

Comment:
It might be more efficient to just do `onEventRef.current = onEvent` instead of adding a useEffect.

Reply:
Good point! Updated